### PR TITLE
fix: NavItem and DropDownItem rendering icon without overriding className

### PR DESCRIPTION
**Pull request by myNameIsDu**
In both NavItem and DropdownItem, the className attribute assigned to the icon prop will be overridden.

image

notice:  there's an extra test case change, but I think it was wrong before.

### CODE REVIEW

File: src/Dropdown/test/DropdownItemSpec.tsx [hunk 1]
```diff
@@ -87,7 +87,18 @@ describe('<Dropdown.Item>', () => {
       </Dropdown>
     );
 
-    expect(screen.getByTestId('item')).to.contain(screen.getByTestId('icon'));
+    expect(screen.getByTestId('icon')).to.contain(screen.getByTestId('icon'));
```
Comment (simonguo) [inline src/Dropdown/test/DropdownItemSpec.tsx:90]: icon contains icon is logically incorrect, it is recommended to use getByRole('treeitem) instead.

**Review by simonguo — CHANGES_REQUESTED** [review]
Missing tests for icon are in NavDropdownItem, NavbarDropdownItem, SidenavDropdownItem, SidenavItem components

File: src/Dropdown/test/DropdownItemSpec.tsx (hunk 1 above)
Comment (myNameIsDu) [reply src/Dropdown/test/DropdownItemSpec.tsx]: Done, there's no need to make any modifications to this change.

Comment (simonguo) [reply src/Dropdown/test/DropdownItemSpec.tsx]: `Unable to find an accessible element with the role "menuitem"` error is caused by the DropdownItem not being visible, you need to set an open attribute for the Dropdown.
```js
it('Should render a icon without overriding className', () => {
    render(
      <Dropdown open>
        <DropdownItem icon={<div className="custom-class" />} />
      </Dropdown>
    );
    expect(screen.getByRole('menuitem')).to.contain('.custom-class');
  });
```

According to the testing-library's suggestion, we prefer to use `getByRole` to get the element.
https://testing-library.com/docs/queries/about#priority

Comment (myNameIsDu) [reply src/Dropdown/test/DropdownItemSpec.tsx]: done

**Comment by myNameIsDu** [comment]
> Missing tests for icon are in NavDropdownItem, NavbarDropdownItem, SidenavDropdownItem, SidenavItem components

done

**Review by simonguo — APPROVED** [review]

**Pull request merged**